src/hardware/gwinstek-gds-800/protocol.c

```diff
@@ -223,17 +235,11 @@ SR_PRIV int gwinstek_gds_800_receive_data(int fd, int revents, void *cb_data)
 			break;
 
 		/* Fetch data needed for conversion from device. */
-		snprintf(command, sizeof(command), ":CHAN%d:SCAL?",
-				devc->cur_acq_channel + 1);
-		if (sr_scpi_get_string(scpi, command, &response) != SR_OK) {
+		if(gwinstek_gds_800_fetch_volts_per_div(sdi->conn, devc->cur_acq_channel, &volts_per_division) != SR_OK) {
 			sr_err("Failed to get volts per division.");
 			sr_dev_acquisition_stop(sdi);
 			return TRUE;
 		}
-		volts_per_division = g_ascii_strtod(response, &end_ptr);
-		if (!strcmp(end_ptr, "mV"))
-			volts_per_division *= 1.e-3;
-		g_free(response);
```

Comment:
You remove this "mV" check, but evidently GDS-800 series models need this, so this is a regression.

Reply:
I'll put this back in. From the GDS-800 documentation I'm not certain it's actually needed but I don't have one to test either way.